torch/package/package_exporter.py

```diff
@@ -970,6 +974,15 @@ def relied_on_by(self, module_name: str) -> List[str]:
         else:
             return []
 
+    def failed_dunder_import_list(self) -> List[str]:
```

Comment:
I'm not sure what discoverability for this API will be. Ideally we'd like the API to be "obvious"—someone will look in the API reference pretty and quickly find what they're looking for. The methods introduced in the PR below are good examples of obvious APIs—they correspond to actions that are common in the rest the API and it's pretty clear what they do even from the name.

But `failed_dunder_import_list` is pretty inscrutable—I'm not sure how a user would figure out what this method does, or how they would know to call it, or what they would do with results. Maybe we should just eat the `__import__` error and call it a day?